### DISCUSSION TITLE
Grid Template Area support?

### DISCUSSION THREAD

**Discussion by cybervaldez** [tailwindlabs/tailwindcss#2634] [Help]
Is this in the pipeline? I’m curious how this will be implemented.

**Comment by simonswiss** [comment]
My gut feeling is that `grid-template-areas` because it's so _bespoke_ and custom that it would be hard to abstract away in a set of useful utility classes.

But... I'd love to be proven wrong 😅

**Reply by cybervaldez** [reply]
I got so spoiled with tailwind 😂 I spent the day looking for the **tailwind way** of doing grid-template-areas even though as you said it is very abstract.

**Reply by brandonpittman** [reply]
> My gut feeling is that `grid-template-areas` because it's so _bespoke_ and custom that it would be hard to abstract away in a set of useful utility classes.
> 
> But... I'd love to be proven wrong 😅

@simonswiss 

It needs to be handled similarly to animations—you provide a list of area names and the build process generates matching classes. You just need to provide the definition.

```js

theme: {
	templateAreas: [
		blog: `
			"head head"
        	"nav  main"
            "nav  foot";
		`
	]
}
```

The build process parses out the space-separated area names, generates those classes as `area-head` and the like while creating a `template-blog` class to use at the top.

**Reply by y1n0** [reply]
That is true, but now with JIT and arbitrary values, it would be nice to be able to use area directly in classnames, even if it does not any preset values in config. I am thinking of something like `class="area-[1/3/3/4]"` given that I have already declared the template with something like `grid-cols-[1fr,1fr,2fr]`. Or if a named a area has been declared: `area-[spotX]`

**Comment by euthiago** [comment]
Support for grid-template-areas is the one BIG missing feature for me.
@container and grid-template-areas together are spectacullar and you **cannot** achieve the same results just using flex.
Flex requires extra "line container" elements for "2d" structures that grids can produce out ot the box.
There are many scenarios that this extra "line container" would need to be removed, moved or have children elements added/removed, forcing the browser back to a much more expensive layout calculation compared to just changing the grid area.
Naming areas is also considerably better then relying on the tipical "start / end" notation.
